src/registry/Instantiator.ts

```diff
@@ -0,0 +1,31 @@
+import Type from "@/type/Type";
+
+class Instantiator {
+
+	public static create<T>(classInstance: Type<T>): (...args: any[]) => T {
+		const fn: (...args: any[]) => T = (...args: any[]) => {
+			if (args.length === 0) {
+				return new classInstance();
+			}
+
+			let argumentsCode: string = "";
```

Comment:
Use of Arrays.join is going to be less error prone and has been fully supported since v1 of most browsers over 20 years.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join